packages/vue-app/template/utils.js

```diff
@@ -193,13 +196,20 @@ export async function setContext(app, context) {
   app.context._redirected = false
   app.context._errored = false
   app.context.isHMR = !!context.isHMR
-  if (context.route) {
-    app.context.route = await getRouteData(context.route)
-  }
   app.context.params = app.context.route.params || {}
```

Comment:
params and query need to be set after route data fetched

Reply:
Oh, sure 😅